check.ts

```diff
@@ -0,0 +1,140 @@
+import { Request, Response } from 'express';
+
+import ReviewService from '@/services/reviewService';
+import type { PullRequestWebhookEvent, PullRequestReviewWebhookEvent, IssueCommentWebhookEvent } from '@/types';
+import config from '@/utils/config';
+import logger from '@/utils/logger';
+
+class WebhookController {
+  private reviewService: ReviewService;
+
+  constructor() {
+    this.reviewService = new ReviewService();
+  }
+
+  async handlePullRequest(req: Request<{}, {}, PullRequestWebhookEvent>, res: Response): Promise<void> {
+    try {
+      const { action, number, pull_request, repository } = req.body;
+
+      logger.info('Received PR webhook', {
+        action,
+        number,
+        repo: repository?.full_name,
+      });
+
+      // Only process certain actions
+      const reviewActions = ['opened', 'synchronize', 'reopened'];
+      if (!reviewActions.includes(action)) {
+        logger.info('Ignoring PR action', { action });
+        return res.status(200).json({ message: 'Action ignored' });
+      }
+
+      // Only process PRs for configured repository
+      // const owner = repository.owner.login;
+      // const repo = repository.name;
+
+      // if (owner !== config.GITHUB_OWNER || repo !== config.GITHUB_REPO) {
+      //   logger.info('Ignoring PR from different repository', {
+      //     owner,
+      //     repo,
+      //     configured: `${config.GITHUB_OWNER}/${config.GITHUB_REPO}`,
+      //   });
+      //   return res.status(200).json({ message: 'Repository not configured for review' });
+      // }
+
+      this.processReviewAsync(repository.owner.login, repository.name, number);
```

Comment:
This async operation is started but not awaited, which means the response is sent immediately without waiting for the review to complete. This could lead to race conditions or make error handling difficult. Consider either awaiting this operation or clearly documenting this fire-and-forget behavior.

```suggestion
await this.processReviewAsync(repository.owner.login, repository.name, number);
```